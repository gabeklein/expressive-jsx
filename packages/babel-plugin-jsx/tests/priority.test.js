transform("competing style priority", `
  () => {
    Bar: { color: blue }
    Foo: {
      Bar: { color: orange }
    }
    
    <Foo>
      <Bar />
    </Foo>
  };
`);

transform("conditional priority should be higher", `
  () => {
    if(checked)
      test: {
        color: red;
      }

    test: {
      color: green;
    }

    <test />
  }
`);

transform("prioritize external (capital-letter) define", `
  const Foo = () => {
    forward: className;
    color: red;

    <this>Hello</this>
  }

  const Bar = () => {
    Foo: {
      color: blue;
    }
    
    <Foo />
  }
`);