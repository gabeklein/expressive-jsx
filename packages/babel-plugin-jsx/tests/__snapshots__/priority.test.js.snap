// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`competing style priority 1`] = `
  () => (
    <Foo>
      <Bar className="Bar_lt8 Bar_1nq" />
    </Foo>
  );
  
  css(\`
  	.Bar_1nq {
  		color: blue;
  	}
  	.Bar_lt8 {
  		color: orange;
  	}
  \`);
`;

exports[`conditional priority should be higher 1`] = `
  () => (
    <div className={!!checked && "element_3ld"}>
      <div className="test_7au test_1gc" />
    </div>
  );
  
  css(\`
  	.test_7au {
  		color: green;
  	}
  	.element_3ld .test_1gc {
  		color: red;
  	}
  \`);
`;

exports[`prioritize external (capital-letter) define 1`] = `
  const Foo = ({ className }) => (
    <div className={classNames("Foo_2t4", className)}>
      Hello
    </div>
  );
  
  
  const Bar = () => <Foo className="Foo_55p" />;
  css(\`
  	.Foo_2t4 {
  		color: red;
  	}
  	.Foo_55p {
  		color: blue;
  	}
  \`);
`;
